aten/src/ATen/native/native_functions.yaml

```diff
@@ -105,6 +105,13 @@
   manual_kernel_registration: True
   variants: method
 
+- func: fw_primal(Tensor(a) self, int level) -> Tensor(a)
+  use_c10_dispatcher: full
+  manual_kernel_registration: True
```

Comment:
how come manual?

Reply:
This has to be manual to be able to do a special forward mode AD update.
I will add a comment mentioning it here.

Reply:
This is the wrong way to suppress AD generation. As you noticed, `manual_kernel_registration` will also suppress the regular Python binding generation, but I'm guessing you didn't make any modifications to the Python binding code.  `MANUAL_AUTOGRAD_AND_TRACER` in `tools/autograd/gen_variable_type.py`  is the correct way to prevent VariableType from being generated.

cc @smessmer

Reply:
Ho I didn't knew about `MANUAL_AUTOGRAD_AND_TRACER`
But yes the python binding needs to be skipped as well. `manual_cpp_binding` is the right way to disable all the python binding? Or is there another flag?

Reply:
Here's the way to think about this. When you add a function to `native_functions.yaml`, here is what you get:

* Schema definition, aka TorchScript support
* C++ bindings in Functions.h and TensorMethods.h (disable with `manual_cpp_binding`), including RecordFunction profiling support
* Dispatch kernel registrations to the schema (disable with `manual_kernel_registration`)
* Python bindings (disable with `SKIP_PYTHON_BINDINGS`)
* Autograd and tracing kernel registrations (disable with `MANUAL_AUTOGRAD_AND_TRACER`, also indirectly disabled by the conjunction of `manual_kernel_registration` and `MANUAL_BACKEND`)

The very first question you should ask is, "does this make sense in TorchScript?" Because if an operator doesn't make sense in TorchScript, chances are it doesn't belong in `native_functions.yaml` at all. (By the way, disabling Python bindings in this way is a code smell, because if automatic Python bindings don't work, it's unlikely the default TorchScript bindings will work--this is a big problem for fast path native functions ala #30774. And in this PR, I see you are disabling the Python bindings, which implies that forward mode AD isn't going to work with TorchScript. There's nothing intrinsically wrong with deciding not to support TorchScript at the beginning, but it is worth trying to avoid making design decisions that make your life harder in the future.)